Source/player.cpp

```diff
@@ -2105,6 +2105,37 @@ void DropHalfPlayersGold(int pnum)
 	plr[pnum]._pGold = CalculateGold(pnum);
 }
 
+#ifdef HELLFIRE
+void StripTopGold(int pnum)
+{
+	ItemStruct tmpItem;
+	int i, val;
+
+	if ((DWORD)pnum >= MAX_PLRS) {
+		app_fatal("StripTopGold: illegal player %d", pnum);
+	}
+	tmpItem = plr[pnum].HoldItem;
+
+	for (i = 0; i < plr[pnum]._pNumInv; i++) {
+		if (plr[pnum].InvList[i]._itype == ITYPE_GOLD) {
+			if (plr[pnum].InvList[i]._ivalue > MaxGold) {
+				val = plr[pnum].InvList[i]._ivalue - MaxGold;
+				plr[pnum].InvList[i]._ivalue = MaxGold;
+				SetGoldCurs(pnum, i);
+				SetPlrHandItem(&plr[pnum].HoldItem, 0);
+				GetGoldSeed(pnum, &plr[pnum].HoldItem);
+				SetPlrHandGoldCurs(&plr[pnum].HoldItem);
+				plr[pnum].HoldItem._ivalue = val;
+				if (!GoldAutoPlace(pnum))

```

Comment:
Personal preference, I prefer to always insert `{` and `}` braces for if-bodies (even one-line bodies). Other than that, good PR!

Reply:
rest of the code has 1line ifs without the braces so I'm trying to follow the style :P

Reply:
> rest of the code has 1line ifs without the braces so I'm trying to follow the style :P

Understood.